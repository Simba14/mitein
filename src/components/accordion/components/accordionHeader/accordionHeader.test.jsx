import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { noop } from 'lodash';
import AccordionHeader from './accordionHeader';

describe('AccordionHeader', () => {
  describe('when is selected', () => {
    it('should render correctly', () => {
      const rendered = renderer.create(
        <AccordionHeader isOpen onClick={noop} ariaId="button-01" ariaControls="panel-01">
          <div>this is the header</div>
        </AccordionHeader>,
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

  describe('when is not selected', () => {
    it('should render correctly', () => {
      const rendered = renderer.create(
        <AccordionHeader isOpen={false} onClick={noop} ariaId="button-01" ariaControls="panel-01">
          <div>this is the header</div>
        </AccordionHeader>,
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

  describe('when clicking', () => {
    it('should call "onClick"', () => {
      const onClick = jest.fn();

      const wrapper = shallow(
        <AccordionHeader isOpen onClick={onClick} ariaId="button-01" ariaControls="panel-01">
          <div>this is the header</div>
        </AccordionHeader>,
      );

      wrapper.find('.accordionHeader').simulate('click');
      expect(onClick).toHaveBeenCalled();
    });
  });
});
