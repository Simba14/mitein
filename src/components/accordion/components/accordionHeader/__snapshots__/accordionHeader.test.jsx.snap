// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionHeader when is not selected should render correctly 1`] = `
<h3>
  <button
    aria-controls="panel-01"
    aria-expanded={false}
    className="accordionHeader"
    id="button-01"
    onClick={[Function]}
  >
    <div>
      this is the header
    </div>
  </button>
</h3>
`;

exports[`AccordionHeader when is selected should render correctly 1`] = `
<h3>
  <button
    aria-controls="panel-01"
    aria-expanded={true}
    className="accordionHeader isOpen"
    id="button-01"
    onClick={[Function]}
  >
    <div>
      this is the header
    </div>
  </button>
</h3>
`;
